src/ripple/conditions/impl/Der.h

```diff
@@ -0,0 +1,2798 @@
+    }
+
+    static
+    std::uint8_t
+    numUnusedBits(
+        std::bitset<S> const& s,
+        std::size_t leadingZeroBytes)
+    {
+        // b is first non-zero byte
+        auto const bits = s.to_ulong ();
+        std::uint8_t const b =
+            (bits >> (maxBytes - leadingZeroBytes - 1) * 8) & 0xff;
+        if (b & 0x80)
+            return 0;
+        if (b & 0x40)
+            return 1;
+        if (b & 0x20)
+            return 2;
+        if (b & 0x10)
+            return 3;
+        if (b & 0x08)
+            return 4;
+        if (b & 0x04)
+            return 5;
+        if (b & 0x02)
+            return 6;
+        if (b & 0x01)
+            return 7;
+        // der always considers the last bit, even if no bits are set
+        return 7;
+    }
+
+    static
+    void
+    encode(Encoder& encoder, std::bitset<S> const& s)
+    {
+        static_assert(
+            maxBytes > 0 && maxBytes <= sizeof(unsigned long),
+            "Unsupported bitset size");
+
+        auto& parentSlice = encoder.parentSlice();
+        auto const bits = s.to_ulong();
+
+        if (bits == 0)
+        {
+            if (parentSlice.size() != 2)
+            {
+                encoder.ec_ = make_error_code(Error::logicError);
+                return;
+            }
+            parentSlice.push_back(7);
+            parentSlice.push_back(0);
+            return;
+        }
+
+        std::size_t const leadingZeroBytes = numLeadingZeroBytes(s);
+        std::uint8_t const unusedBits = numUnusedBits(s, leadingZeroBytes);
+
+        if (parentSlice.size() != 1 + maxBytes - leadingZeroBytes)
+        {
+            encoder.ec_ = make_error_code(Error::logicError);
+            return;
+        }
+
+        parentSlice.push_back(unusedBits);
+
+        for (size_t curByte = 0; curByte < maxBytes - leadingZeroBytes; ++curByte)
+        {
+            uint8_t const v = (bits >> curByte * 8) & 0xff;
+            parentSlice.push_back(reverseBits(v));
+        }
+    }
+
+    static
+    void
+    decode(Decoder& decoder, std::bitset<S>& v)
+    {
+        static_assert(
+            maxBytes > 0 && maxBytes <= sizeof(unsigned long),
+            "Unsupported bitset size");
+
+        auto& slice = decoder.parentSlice();
+        std::error_code& ec = decoder.ec_;
+
+        if (slice.empty() || slice.size() > maxBytes + 1)
+        {
+            ec = make_error_code(Error::contentLengthMismatch);
+            return;
+        }
+
+        auto const unused = slice[0];
+        slice += 1;
+
+        if (unused < minUnusedBits)
+        {
+            ec = make_error_code(Error::contentLengthMismatch);
+            return;
+        }
+
+        if (unused >= 8)
+        {
+            ec = make_error_code(Error::badDerEncoding);
+            return;
+        }
+
+        unsigned long bits = 0;
+        auto const numBytes = slice.size();
+        size_t curByteIndex = 0;
+        for (; !slice.empty(); ++curByteIndex, slice += 1)
```

Comment:
`slice` does not define `operator++` (and I don't think it's worth adding just for this).